wgengine/magicsock/endpoint.go

```diff
@@ -354,6 +359,44 @@ func (de *endpoint) addrForWireGuardSendLocked(now mono.Time) (udpAddr netip.Add
 	return udpAddr, needPing
 }
 
+// addrForPingSizeLocked returns the address(es) that should be used for sending
+// the next ping. It will only return addrs with a large enough path MTU to
+// permit a ping payload of size bytes to be delivered (DERP is always one such
+// addr as it is a TCP connection). If it returns a zero-value udpAddr, then we
+// should continue probing the MTU of all paths to this endpoint. Zero, one, or
+// both of the returned UDP address and DERP address may be non-zero.
+//
+// de.mu must be held.
+func (de *endpoint) addrForPingSizeLocked(now mono.Time, size int) (udpAddr, derpAddr netip.AddrPort) {
+	if size == 0 {
+		udpAddr, derpAddr, _ = de.addrForSendLocked(now)
+		return
+	}
+
+	udpAddr = de.bestAddr.AddrPort
+	mtuOk := true
+	pathMTU := de.bestAddr.wireMTU
+	requestedMTU := pingSizeToPktLen(size, udpAddr.Addr())
+	if requestedMTU > pathMTU {
+		mtuOk = false
+	}
```

Comment:
I found this a bit difficult to follow with a bunch of reassignments and temporary variables, may I suggest to simplify a tiny bit? For example:

```suggestion
	pathMTU := de.bestAddr.wireMTU
	requestedMTU := pingSizeToPktLen(size, udpAddr.Addr())
	mtuOk := requestedMTU <= pathMTU
```

Reply:
Thanks, applied!